src/index.js

```diff
@@ -1,260 +1,308 @@
-    const camelized = camelize(key)
-    let value = el.hasAttribute(key) ? el.getAttribute(key) : undefined
+    function syncProperty(el, key, syncJsProp) {
+        const camelized = camelize(key)
+        let value = el.hasAttribute(key) ? el.getAttribute(key) : undefined
 
-    if (syncJsProp) {
-      if (el.props && el.props[camelized] !== undefined) {
-        value = el.props[camelized]
-      } else {
-        value = el[camelized] !== undefined ? el[camelized] : value
-      }
+        if (syncJsProp) {
+            if (el.props && el.props[camelized] !== undefined) {
+                value = el.props[camelized]
+            } else {
+                value = el[camelized] !== undefined ? el[camelized] : value
+            }
+        }
+
+        el._wrapper.props[camelized] = convertAttributeValue(
+            value,
+            key,
+            camelizedPropsMap[camelized]
+        )
     }
 
-    el._wrapper.props[camelized] = convertAttributeValue(
-      value,
-      key,
-      camelizedPropsMap[camelized]
-    )
-  }
+    function syncAttribute(el, key) {
+        if (isIgnoredAttribute(key) || hyphenatedPropsList.indexOf(key) !== -1) {
+            return
+        }
 
-  function syncAttribute (el, key) {
-    if (isIgnoredAttribute(key) || hyphenatedPropsList.indexOf(key) !== -1) {
-      return
-    }
+        const value = el.hasAttribute(key) ? el.getAttribute(key) : undefined
+        const wrapper = el._wrapper
 
-    const value = el.hasAttribute(key) ? el.getAttribute(key) : undefined
-    const wrapper = el._wrapper
+        if (!wrapper || !wrapper._vnode) {
+            return
+        }
 
-    if (!wrapper || !wrapper._vnode) {
-      return
+        wrapper._update(Object.assign({}, wrapper._vnode, {
+            data: Object.assign({}, wrapper._vnode.data, {
+                attrs: Object.assign(
+                    {},
+                    wrapper._vnode.data.attrs,
+                    { [key]: value }
+                )
+            })
+        }), false)
     }
 
-    wrapper._update(Object.assign({}, wrapper._vnode, {
-      data: Object.assign({}, wrapper._vnode.data, {
-        attrs: Object.assign(
-          {},
-          wrapper._vnode.data.attrs,
-          { [key]: value }
-        )
-      })
-    }), false)
-  }
+    class CustomElement extends HTMLElement {
+        constructor() {
+            const self = super();
+            this._el = self;
+            this.props = {};
+            const shadow = self.attachShadow({ mode: 'open', delegatesFocus: delegatesFocus })
+            css && css.forEach(promise =>
+                promise.then(res => res.clone().text().then(content => {
+                    const style = document.createElement('style')
+                    style.appendChild(document.createTextNode(content))
+                    shadow.appendChild(style)
+                }))
+            )
+            this._createWrapper();
+            this._createObserver();
```

Comment:
Are you not creating the observable twice? Once here and once in `_createWrapper` on line 272?

Reply:
is not created twice bc of those ifs in the createObserver function but you are right i believe i can avoid the one on line 132.